packages/autorest.gotest/test/integrationtest/output/signalr/armsignalr/fake_test.go

```diff
@@ -0,0 +1,1718 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+package armsignalr_test
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"reflect"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	azfake "github.com/Azure/azure-sdk-for-go/sdk/azcore/fake"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/internal/v2/testutil"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/signalr/armsignalr"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/signalr/armsignalr/fake"
+	"github.com/stretchr/testify/suite"
+)
+
+type FakeTestSuite struct {
+	suite.Suite
+
+	cred           azcore.TokenCredential
+	subscriptionId string
+}
+
+func (testsuite *FakeTestSuite) SetupSuite() {
+	testsuite.cred = &testutil.FakeCredential{}
+	testsuite.subscriptionId = "00000000-0000-0000-0000-000000000000"
+}
+
+func TestFakeTest(t *testing.T) {
+	suite.Run(t, new(FakeTestSuite))
+}
+
+func (testsuite *FakeTestSuite) TestOperations_List() {
+	ctx := context.Background()
+	fakeServer := fake.OperationsServer{}
+	clientFactory, err := armsignalr.NewClientFactory(testsuite.subscriptionId, testsuite.cred, &arm.ClientOptions{
```

Comment:
one service's client factory does not need to init multiple times

Reply:
Transport needs to be set up separately for each operation
![image](https://github.com/Azure/autorest.go/assets/46921893/3f16ab03-e3fa-4524-9c96-e7f6af880bf9)

Reply:
@tadelesh @Alancere I know I'm not directly involved in the dev work here (so treat it as a suggestion), but isn't that what `NewServerFactoryTransport` is for? to match `ClientFactory` with `ServerFactory`? like:

```go
Transport: fake.NewServerFactoryTransport(&fakeServerFactory),
```

Example in your repo:
https://github.com/Azure/autorest.go/blob/main/packages/autorest.go/test/compute/armcompute/fakes_test.go#L52